frontend/src/board/board.ts

```diff
@@ -45,5 +47,105 @@ export class Board {
       { row: 0, column: 2},
       { row: 0, column: 3},
     ];
+
+  }
+
+  findMoves() {
+    return {
+      north: this.checkDirections(this.robotPositions[0].row, this.robotPositions[0].column, 'North'),
+    south: this.checkDirections(this.robotPositions[0].row, this.robotPositions[0].column, 'South'),
+    east: this.checkDirections(this.robotPositions[0].row, this.robotPositions[0].column, 'East'),
+    west: this.checkDirections(this.robotPositions[0].row, this.robotPositions[0].column, 'West')
+    }
+  }
+
+  checkDirections(startRow:number, startCol: number, direction: 'North' | 'South' | 'East' | 'West' ) : null | Position {
+    // Gets the position of the target robot
+    // Checks the north, south, east, and west path
+    // Path ends if
+    // - it is the last cell in the direction
+    // - or the cell is obstructed with either another robot, a wall, or the center pieces
+    // - or the cell is the target cell
+
+    let legalMove: Position | null = null 
+    
+    
+    let row = startRow
+    let col = startCol
+ 
+    while (row >= 0 && row < BOARD_SIZE && col >= 0 && col < BOARD_SIZE) {
+
+      // check if target, add position
+
+      let robotEncountered = false 
+      // robots 
+      this.robotPositions.forEach(robot => {
+        // designated robot does not check its own position
+        if (robot.row === startRow && robot.column === startCol) {
+          return 
+        }
+        if (row === robot.row && col === robot.column) {
+          robotEncountered = true 
+        }
+      })
+
+      if (robotEncountered) {
+        break; 
+      }
+
+
+      if (this.cells[row]?.[col]?.isTarget) {
+        legalMove = {row: row, column: col}
+        break
+      }
+
+      
+      // check if obstructed
+      if (this.cells[row]?.[col]?.isObstructed) {
+        break
+      }
+
+      let wallPreventsEntrance = false 
+      // check if wall exists
+      if (this.cells[row]?.[col]?.walls?.length! > 0) {
+  
+        this.cells[row]?.[col]?.walls.forEach(wall => {
+          // Wall prevents entrance into next cell
+          if ((direction === 'North' && wall === Direction.North) || 
+            (direction === 'South' && wall === Direction.South) ||
+            (direction === 'East' && wall === Direction.East) ||
+            (direction === 'West' && wall === Direction.West)) {
+            wallPreventsEntrance = true
+
+            if (!(row === startRow && col === startCol))
+            legalMove = {row: row, column:col}
+          } 
+          
+    
+        })
+      }
+      if (wallPreventsEntrance) {
+        
+        break
+      }
+  
+      
+      if (!(row === startRow && col === startCol)) {
+        legalMove = {row: row, column: col}
+      }
+
+      if (direction === 'North') {
+        row--;
+      } else if (direction === 'South') {
+        row++;
+      } else if (direction === 'East') {
+        col++;
+      } else if (direction === 'West') {
+        col--;
+      }
```

Comment:
As a broad meta level comment I think it might simplify things a little bit if we increment our position before entering the while loop so we can always assume that our current `(row, col)` values are a valid legal move, and break if we discover that the next coordinates are not. Let me know if that doesn't make sense